lib_bgp_data/roas_parser/roas_automator.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""This module contains the ROAs_Automator
+
+The purpose of this class is to parse the roas and insert them into the
+database. This class will properly check to see if the database needs to be
+restored and will create a backup each time the _run method is called.
+
+First, the roas table in the database is checked. If this table is empty or
+dne, the table will be restored from a previous backup. If there is no
+previous backup, new data will be inserted using an emppty table.
+
+Next, the roas json is retrieved. The new data is inserted and a new backup is
+made. The table is then restored from the is backup. If number of rows in the
+table is the same after the insertion and then after the restore, the previous
+backup is overwritten. If the tempporary backup doesn't contain the same amount
+of data, we try creating a backup again. We try to make a backup at most three
+times. If a backup cannot be made for some reason, we do not overwrite the
+previous backup.
+
+If the backup fails, an email will be sent, notifying the user of any errors
+if pg_dump or pg_restore have any.
+
+This automator is designed to be run nightly on our server using cron.
+"""
+
+__authors__ = ["Samarth Kasbawala"]
+__credits__ = ["Samarth Kasbawala"]
+__Lisence__ = "BSD"
+__maintainer__ = "Justin Furuness"
+__email__ = "jfuruness@gmail.com"
+__status__ = "Development"
+
+import os
+import psycopg2
+import subprocess
+from .tables import ROAs_Table
+from .roas_parser import ROAs_Parser
+from ..base_classes import Parser
+from ..utils import utils
+
+
+class TableIsEmpty(Exception):
+    pass
+
+class BackupError(Exception):
+    pass
+
+class ROAs_Automator(ROAs_Parser):
+    backup_dir = "/nightly_runs/roas_backup/"
+    temp_backup = f"{backup_dir}roas_temp.sql.gz"
```

Comment:
what will happen if the backup directory does not exist? I think you should create it in the init section